contracts/strategies/donation-voring-merkle-distribution/DonationVotingMerkleDistributionStrategy.sol

```diff
@@ -0,0 +1,602 @@
+    /// ===============================
+    /// ======= External/Custom =======
+    /// ===============================
+
+    /// @notice Review recipient application
+    /// @param _recipientIds Ids of the recipients
+    /// @param _recipientStatuses Statuses of the recipients
+    function reviewRecipients(address[] calldata _recipientIds, InternalRecipientStatus[] calldata _recipientStatuses)
+        external
+        onlyPoolManager(msg.sender)
+        onlyActiveRegistration
+    {
+        uint256 recipientLength = _recipientIds.length;
+        if (recipientLength != _recipientStatuses.length) {
+            revert INVALID();
+        }
+
+        for (uint256 i = 0; i < recipientLength;) {
+            InternalRecipientStatus recipientStatus = _recipientStatuses[i];
+            address recipientId = _recipientIds[i];
+            if (recipientStatus == InternalRecipientStatus.None || recipientStatus == InternalRecipientStatus.Appealed)
+            {
+                revert RECIPIENT_ERROR(recipientId);
+            }
+
+            Recipient storage recipient = _recipients[recipientId];
+
+            recipient.recipientStatus = recipientStatus;
+
+            emit RecipientStatusUpdated(recipientId, recipientStatus, msg.sender);
+
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /// @notice Set the start and end dates for the pool
+    /// @param _registrationStartTime The start time for the registration
+    /// @param _registrationEndTime The end time for the registration
+    /// @param _allocationStartTime The start time for the allocation
+    /// @param _allocationEndTime The end time for the allocation
+    function updatePoolTimestamps(
+        uint256 _registrationStartTime,
+        uint256 _registrationEndTime,
+        uint256 _allocationStartTime,
+        uint256 _allocationEndTime
+    ) external onlyPoolManager(msg.sender) {
+        _isPoolTimestampValid(_registrationStartTime, _registrationEndTime, _allocationStartTime, _allocationEndTime);
+
+        registrationStartTime = _registrationStartTime;
+        registrationEndTime = _registrationEndTime;
+        allocationStartTime = _allocationStartTime;
+        allocationEndTime = _allocationEndTime;
+
+        emit TimestampsUpdated(
+            registrationStartTime, registrationEndTime, allocationStartTime, allocationEndTime, msg.sender
+        );
+    }
+
+    /// @notice Withdraw funds from pool
+    /// @param _amount The amount to be withdrawn
+    function withdraw(uint256 _amount) external onlyPoolManager(msg.sender) onlyAfterAllocation {
+        if (block.timestamp <= allocationEndTime + 30 days) {
+            revert NOT_ALLOWED();
+        }
+
+        // TODO: discuss if we want to allow withdrawing ONLY "poolAmount - totalPayoutAmount" from the pool
+        IAllo.Pool memory pool = allo.getPool(poolId);
+        if (poolAmount - totalPayoutAmount < _amount) {
+            revert NOT_ALLOWED();
+        }
```

Comment:
^ need to discuss this. cause this is for situations where they wanted to pull back the remaining funds due to various reasons:
- user didn't KYC so manager didn;t play 
- user forgot to claim